From b4d9fd8d832c8b82c85424bf57cd660908452f24 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Tue, 6 Oct 2020 14:43:16 -0700
Subject: [PATCH 06/36] UBUNTU: SAUCE: apparmor: disable showing the mode as
 part of a secid to secctx

Displaying the mode as part of the seectx takes up unnecessary memory,
makes it so we can't use refcounted secctx so we need to alloc/free on
every conversion from secid to secctx and introduces a space that
could be potentially mishandled by tooling.

Eg. In an audit record we get

  subj_type=firefix (enforce)

Having the mode reported is not necessary, and might even be confusing
eg. when writing an audit rule to match the above record field you
would use

  -F subj_type=firefox

ie. the mode is not included. AppArmor provides ways to find the mode
without reporting as part of the secctx. So disable this by default
before its use is wide spread and we can't. For now we add a sysctl
to control the behavior as we can't guarentee no one is using this.

Signed-off-by: John Johansen <john.johansen@canonical.com>
Acked-by: Andrea Righi <andrea.righi@canonical.com>
Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 security/apparmor/secid.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/security/apparmor/secid.c b/security/apparmor/secid.c
index 24a0e23f1b2ba..1aa0d61e34550 100644
--- a/security/apparmor/secid.c
+++ b/security/apparmor/secid.c
@@ -33,6 +33,8 @@ static DEFINE_XARRAY_FLAGS(aa_secids, XA_FLAGS_LOCK_IRQ | XA_FLAGS_TRACK_FREE);
 
 int apparmor_display_secid_mode;
 
+int apparmor_display_secid_mode = 0;
+
 /*
  * TODO: allow policy to reserve a secid range?
  * TODO: add secid pinning
-- 
2.38.1

